File: src/my_project/easy_problems/from1to50/pascal_triangle.py
from typing import List

if __name__ == '__main__':
    class Solution:
        def generate(self, numRows: int) -> List[List[int]]:
            temp = list()
            list_rows = [[1],[1,1]]

            if numRows == 1:
                return [[1]]
            elif numRows == 2:
                return [[1], [1,1]]
            elif numRows >= 3:

                list_pascal = [1,1]
                i = 3
                while i <= numRows:

                    for j in range(i):
                        if j ==0:
                            temp.append(list_pascal[0])
                        if j == (i -1):
                            temp.append(list_pascal[-1])
                        elif j > 0 and j < i-1:
                            temp.append(list_pascal[j-1] + list_pascal[j])

                    i +=1
                    list_pascal = temp
                    list_rows.append(list_pascal)
                    temp = list()
                return list_rows


    solution = Solution()
    print(solution.generate(1))
    sample = [1,2,3]
    print(sample[-1])
    temp = sample
    sample = list()
    print(temp)

File: src/my_project_typescript/easy_problems_typescript/from151to200/pascalTriangle.ts
function generate(numRows: number): number[][] {
    let temp: Array<number> = []

    let solution: number[][] = [[1],[1,1]]

    if (numRows===1){
        return [[1]]
    } 
    else if (numRows===2){
        return [[1],[1,1]]
    }
    else{
        let listPascal: number[] = [1,1]
        let i: number = 3

        while (i<=numRows){
            for (let j = 0; j < i; j++){
                if (j === 0){
                    temp.push(listPascal[0])
                }
                else if (j === (i -1)){
                    temp.push(listPascal[listPascal.length-1])
                }
                else if (j > 0 && j < i-1){
                    temp.push(listPascal[j-1]+listPascal[j])
                }
            }
            i += 1
            listPascal = temp 
            solution.push(listPascal)
            temp = []
        }
        return solution
    }




};